magefile.go

```diff
@@ -1295,3 +1320,170 @@ func shouldBuildAgent() bool {
+		return fmt.Errorf("%s CLI is not installed: %w", cliName, err)
+	}
+
+	// Check if user is already authenticated
+	var authList []struct {
+		Account string `json:"account"`
+	}
+	for authSuccess := false; !authSuccess; {
+		cmd = exec.CommandContext(ctx, cliName, "auth", "list", "--filter=status:ACTIVE", "--format=json")
+		output, err := cmd.Output()
+		if err != nil {
+			return fmt.Errorf("unable to list authenticated accounts: %w", err)
+		}
+
+		if err := json.Unmarshal(output, &authList); err != nil {
+			return fmt.Errorf("unable to parse authenticated accounts: %w", err)
+		}
+
+		if len(authList) > 0 {
+			// We have at least one authenticated, active account. All set!
+			authSuccess = true
+			continue
+		}
+
+		fmt.Fprintln(os.Stderr, "❌  GCP authentication unsuccessful. Retrying...")
+
+		// Try to authenticate user
+		cmd = exec.CommandContext(ctx, cliName, "auth", "login")
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("unable to authenticate user: %w", cliName, err)
+		}
+	}
+
+	// Skip additional Elastic-specific checks if so requested
+	const skipEnvVar = "TEST_INTEG_AUTH_GCP_SKIP_ELASTIC_CHECK"
+	skipElasticChecks := os.Getenv(skipEnvVar)
+	if skipElasticChecks != "" {
+		shouldSkip, err := strconv.ParseBool(skipElasticChecks)
+		if err != nil {
+			return fmt.Errorf("unable to parse environment variable %s as boolean: %w", skipEnvVar, err)
+		}
+
+		if shouldSkip {
+			// We're all set, no further checks to do.
+			return nil
+		}
+	}
+
+	// Check that authenticated account's email domain name
+	const expectedEmailDomain = "elastic.co"
+	email := authList[0].Account
+	parts := strings.Split(email, "@")
+	if len(parts) != 2 || parts[1] != expectedEmailDomain {
+		return fmt.Errorf("please authenticate with your @%s email address (currently authenticated with %s)", expectedEmailDomain, email)
+	}
+
+	// Check the authenticated account's project
+	cmd = exec.CommandContext(ctx, cliName, "config", "get", "core/project")
+	output, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("unable to get project: %w", err)
+	}
+
+	const expectedProject = "elastic-platform-ingest"
+	project := strings.TrimSpace(string(output))
+	if project == expectedProject {
+		// We're all set!
+		return nil
+	}
+
+	// Attempt to select correct GCP project
+	fmt.Printf("Attempting to switch GCP project from [%s] to [%s]...\n", project, expectedProject)
+	cmd = exec.CommandContext(ctx, cliName, "config", "set", "core/project", expectedProject)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Stdin = os.Stdin
+	if err = cmd.Run(); err != nil {
+		return fmt.Errorf("unable to switch project from [%s] to [%s]: %w", project, expectedProject, err)
+	}
+
+	return nil
+}
+
+func authESS(ctx context.Context) error {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("unable to determine user's home directory: %w", err)
+	}
+	essAPIKeyFile := filepath.Join(homeDir, ".config", "ess", "api_key.txt")
```

Comment:
Is this location portable? (I don't know if it's the right place to store it in Windows)

Reply:
It should be because `homeDir` itself should be portable (see [`os.UserHomeDir()`](https://pkg.go.dev/os#UserHomeDir)).